test/libsolidity/SemanticTest.cpp

```diff
@@ -92,18 +91,31 @@ SemanticTest::SemanticTest(
 		m_shouldRun = false;
 
 	auto const eofEnabled = solidity::test::CommonOptions::get().eofVersion().has_value();
-	std::string compileViaYul = m_reader.stringSetting("compileViaYul", eofEnabled ? "true" : "also");
 
-	if (compileViaYul == "false" && eofEnabled)
+	auto const compileViaYul = m_reader.enumSetting<CompileViaYul>(
+		"compileViaYul",
+		{
+			{toString(CompileViaYul::False), CompileViaYul::False},
+			{toString(CompileViaYul::True), CompileViaYul::True},
+			{toString(CompileViaYul::Also), CompileViaYul::Also},
+			{toString(CompileViaYul::onlyOnEOF), CompileViaYul::onlyOnEOF}
+		},
+		eofEnabled ? toString(CompileViaYul::True) : toString(CompileViaYul::Also)
+	);
+
+	std::string bytecodeFormatString = m_reader.stringSetting("bytecodeFormat", "unset");
+	if (bytecodeFormatString == ">=EOFv1" && compileViaYul == CompileViaYul::False)
```

Comment:
You're not using the newly added getter.

Reply:
Also, like I said in https://github.com/ethereum/solidity/pull/15819#discussion_r1952990302, the getter should expose already parsed value, not the raw setting. So it should work like this:
```suggestion
	if (bytecodeFormat().count(BytecodeFormat::EOFv1) != 0 && compileViaYul == CompileViaYul::False)
```

I.e. you still use `stringSetting` to define it in `TestCase` and "parsing" it is just recognizing a few hard-coded values, but the end result is a proper set of enum values. The test cases can then use that without having to assume what kind of syntax is supported in the setting. For example in the future we'll want to support a full range of comparison operators, not just `>=`, but we shouldn't really have to care about that here - here we just want to know if the final value includes EOF or not.